data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -354,11 +355,20 @@ private void commitOffsets(boolean forceCommit) {
             offsetsToCommit.forEach(((partition, offset) -> updateCommitCountMetric(partition, offset)));
             try {
                 consumer.commitSync(offsetsToCommit);
+                lastCommitTime = currentTimeMillis;
+            } catch (final RebalanceInProgressException ex) {
+                LOG.error("Failed to commit offsets in topic {} due to rebalance in progress", topicName, ex);
+                try {
+                    Thread.sleep(100);
+                } catch (final InterruptedException ie) {
+                    throw new RuntimeException("Interrupted while waiting to retry after failing to commit offsets");
+                }
+                return;
             } catch (Exception e) {
```

Comment:
Currently, this exception set the lastCommitTime. But, now we don't. Is that ok?

Reply:
@dlvenable it is correct because the last commit time should be set only if commit really happens.